core/modules/replica/ClusterHealthJob.cc

```diff
@@ -0,0 +1,265 @@
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/ClusterHealthJob.h"
+
+// System headers
+#include <stdexcept>
+#include <thread>
+
+// Qserv headers
+#include "lsst/log/Log.h"
+#include "replica/Controller.h"
+#include "replica/QservMgtServices.h"
+#include "replica/ServiceProvider.h"
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.ClusterHealthJob");
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+// ----------------------
+//  Class: ClusterHealth
+// ----------------------
+
+ClusterHealth::ClusterHealth(std::vector<std::string> const& workers)
+    :   _good(false) {
+
+    for (auto const& worker: workers) {
+        _replication[worker] = false;
+        _qserv      [worker] = false;
+    }
+}
+
+void ClusterHealth::updateReplicationState(std::string const& worker,
+                                           bool state) {
+    _replication[worker] = state;
+    updateSummaryState();
+}
+
+void ClusterHealth::updateQservState(std::string const& worker,
+                                     bool state) {
+    _qserv[worker] = state;
+    updateSummaryState();
+}
+
+void ClusterHealth::updateSummaryState() {
+    bool updatedState = true;
+    for (auto&& entry: _replication) {
+        updatedState = updatedState and entry.second;
+    }
+    for (auto&& entry: _qserv) {
+        updatedState = updatedState and entry.second;
+    }
+    _good = updatedState;
+}
+
+Job::Options const& ClusterHealthJob::defaultOptions() {
+    static Job::Options const options{
+        3,      /* priority */
```

Comment:
kind of a worrisome magical number...

Reply:
Yes, it is until I decide to push those (defaults) numbers into the Configuration.